### PR TITLE
Fix salt-ssh pillar bug for state.sls_id function

### LINKED ISSUE #51353: state.sls_id fails to read pillar data from CLI arg
### Description of Issue/Question

`state.sls_id` ignores pillar data provided as CLI argument:

```
$ salt-ssh 'myserver' state.sls_id test_file test pillar='{"test_file": "123"}'
myserver:
----------
          ID: test_file
    Function: file.managed
        Name: /root/.test_file
      Result: False
     Comment: Pillar test_file does not exist
     Started: 10:37:04.734519
    Duration: 2.125 ms
     Changes:   
```

While `state.apply` reads it correctly:

```
$ salt-ssh 'myserver' state.apply test pillar='{"test_file": "123"}'
myserver:
----------
          ID: test_file
    Function: file.managed
        Name: /root/.test_file
      Result: True
     Comment: File /root/.test_file updated
     Started: 10:36:32.526014
    Duration: 16.789 ms
     Changes:   
              ----------
              diff:
                  New file
```

Documentation says that `pillar` argument is supported by `state.sls_id`:

https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.state.html#salt.modules.state.sls_id

### Setup

**test.sls**:

```
test_file:
  file.managed:
    - name: /root/.test_file
    - contents_pillar: test_file
```

### Steps to Reproduce Issue

Run `state.sls_id` with salt-ssh and set pillar data at the command line.

### Versions Report

```
$ salt-ssh --versions-report
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: 2.7.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.30.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.0
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 3.7.3
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15 (default, Jan 10 2019, 23:20:52)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 17.1.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 5.1.1
            ZMQ: 4.3.1
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 4.20.3-arch1-1-ARCH
         system: Linux
        version: Not Installed
 ```

### PR DESCRIPTION
**Pull request by xuhcc**
### What does this PR do?

Fixes #51353

### Previous Behavior

salt-ssh ignored `pillar` CLI argument during execution of `state.sls_id` function

### New Behavior

now it should work

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Review by Ch3LL — COMMENTED** [review]
thanks for pushing the fix for this :) really appreciate it. any chance we can get a test added here to insure this does not regress in the future? Should be pretty easy to use the current state.sls_id tests in tests/integration/ssh/test_state.py for an idea on how to approach the test.

**Comment by xuhcc** [comment]
@Ch3LL Unfortunately I can't get integration tests working. 

```
$ python tests/runtests.py -n integration.ssh.test_state.SSHStateTest.test_state_sls_id
 --------  Skipped Tests  ------------------------
   -> integration.ssh.test_state.SSHStateTest.test_state_sls_id  ->  SSH tests are disabled
```

**Comment by Ch3LL** [comment]
what if you add the arg `--ssh`

**Comment by Ch3LL** [comment]
FYI I migrated this PR from develop to neon to ensure it is included in the upcoming neon release. Let me know if this caused any issues. Thanks

**Comment by xuhcc** [comment]
@Ch3LL `--ssh` worked, thank you! I added the test.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/integration/ssh/test_state.py [hunk 1]
```diff
@@ -83,6 +83,19 @@ def test_state_sls_wrong_id(self):
...
+        test state.sls_id with pillar data
+        '''
+        self.run_function('state.sls_id', [
+            'ssh-file-test',
+            SSH_SLS,
+            'pillar=\'{"test_file_name": "test_pillar"}\'',
+        ])
+        check_file = self.run_function('file.file_exists', ['/tmp/test_pillar'])
+        self.assertTrue(check_file)
+        os.remove('/tmp/test_pillar')
```
[the remark this answers is not in the document]
Comment (xuhcc) [reply tests/integration/ssh/test_state.py]: Done.

**Review by waynew — DISMISSED** [review]

**Comment by waynew** [comment]
re-run all

**Comment by sagetherage** [comment]
@xuhcc will you rebase against master, please?

**Comment by xuhcc** [comment]
@sagetherage Done!

**Comment by Ch3LL** [comment]
@xuhcc looks like there is a merge conflict here.

**Comment by Ch3LL** [comment]
looks like just a lint and pre-commit failure now.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**